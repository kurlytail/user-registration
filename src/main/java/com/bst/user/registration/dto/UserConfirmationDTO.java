
package com.bst.user.registration.dto;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import com.bst.user.registration.constraints.ValidConfirmationHash;

@ValidConfirmationHash
public class UserConfirmationDTO {

	@Email
	@NotNull
	@NotEmpty
	private String email;

	@NotNull
	@NotEmpty
	private String hash;

	public String getHash() {
		return hash;
	}

	public void setHash(String hash) {
		this.hash = hash;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

}
