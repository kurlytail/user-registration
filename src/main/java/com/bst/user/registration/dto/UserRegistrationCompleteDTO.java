package com.bst.user.registration.dto;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import com.bst.user.registration.constraints.ValidConfirmationHash;


@ValidConfirmationHash
public class UserRegistrationCompleteDTO {

	@Email
	@NotNull
	@NotEmpty
	private String email;
	
	@NotNull
	@NotEmpty
	private String name;

	@NotNull
	@NotEmpty
	private String password;
	
	@NotNull
	@NotEmpty
	private String confirmPassword;
	
	@NotNull
	@NotEmpty
	private String hash;

	public String getHash() {
		return hash;
	}

	public void setHash(String hash) {
		this.hash = hash;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getConfirmPassword() {
		return confirmPassword;
	}

	public void setConfirmPassword(String confirmPassword) {
		this.confirmPassword = confirmPassword;
	}
}
