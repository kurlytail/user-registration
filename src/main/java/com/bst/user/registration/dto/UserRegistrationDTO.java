package com.bst.user.registration.dto;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import com.bst.utility.constraints.ValidReCaptcha;

public class UserRegistrationDTO {
     
    @Email
    @NotNull
    @NotEmpty
    private String email;
    
    @ValidReCaptcha
    private String reReCaptchaResponse;

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getReCaptchaResponse() {
		return reReCaptchaResponse;
	}

	public void setReCaptchaResponse(String reReCaptchaResponse) {
		this.reReCaptchaResponse = reReCaptchaResponse;
	}

}
