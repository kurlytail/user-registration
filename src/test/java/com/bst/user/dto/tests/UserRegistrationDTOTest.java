package com.bst.user.dto.tests;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import com.bst.user.registration.dto.UserRegistrationDTO;

import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;
import static io.github.jsonSnapshot.SnapshotMatcher.expect;

public class UserRegistrationDTOTest {
	
	@BeforeClass
	public static void startSnapshot() {
		start();
	}
	
	@AfterClass
	public static void stopSnapshot() {
		validateSnapshots();
	}

	@Test
	public void testDefaultConstruction() {
		UserRegistrationDTO userDTO = new UserRegistrationDTO();
		expect(userDTO).toMatchSnapshot();
	}
	
	@Test
	public void testFields() {
		UserRegistrationDTO userDTO = new UserRegistrationDTO();
		userDTO.setEmail("some-email");
		userDTO.setReCaptchaResponse("some-captcha");
		expect(userDTO).toMatchSnapshot();
	}

}
