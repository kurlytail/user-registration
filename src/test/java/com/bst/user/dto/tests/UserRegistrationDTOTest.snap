com.bst.user.registration.dto.tests.UserRegistrationDTOTest.testDefaultConstruction=[
  {}
]


com.bst.user.registration.dto.tests.UserRegistrationDTOTest.testFields=[
  {
    "email": "some-email",
    "reReCaptchaResponse": "some-captcha"
  }
]